crates/ide-assists/src/handlers/convert_let_else_to_match.rs

```diff
@@ -0,0 +1,441 @@
+use syntax::ast::{edit::AstNodeEdit, AstNode, HasName, LetStmt, Pat};
+use syntax::T;
+
+use crate::{AssistContext, AssistId, AssistKind, Assists};
+
+/// Gets a list of binders in a pattern, and whether they are mut.
+fn binders_in_pat(pat: &Pat) -> Option<Vec<(String, bool)>> {
+    use Pat::*;
+    match pat {
+        IdentPat(p) => {
+            let ident = p.name()?.text().to_string();
```

Comment:
There is no need to eagerly `to_string` this. You can just pass around the `ast::NameRef` until you need the string version.